assets/aml-benchmark/components/src/dataset_preprocessor/dataset_preprocessor.py

```diff
@@ -0,0 +1,136 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ---------------------------------------------------------
+
+"""DataPreprocessor class and runner."""
+
+import json
+import os
+import re
+import jinja2
+import sys
+from utils.exceptions import BenchmarkValidationException
+from utils.error_definitions import BenchmarkValidationError
+from utils.logging import get_logger
+from utils.io import read_jsonl_files
+from azureml._common._error_definition.azureml_error import AzureMLError
+
+current_folder = os.path.dirname(os.path.abspath(__file__))
+parent_folder = os.path.dirname(current_folder)
+sys.path.append(parent_folder)
+logger = get_logger(__name__)
+
+jinja2.filters.FILTERS['zip'] = zip
+ENV = jinja2.Environment()
+ENV.globals.update(zip=zip)
+
+
+class DatasetPreprocessor(object):

```

Comment:
Do we expect users to consume this class via code? Or will it always be a part of the component?

Reply:
After discussing the new design, it will be a part of the component only.